dojo/models.py

```diff
@@ -2251,13 +2251,16 @@ def hash_code_allows_null_cwe(self):
         deduplicationLogger.debug(f"HASHCODE_ALLOWS_NULL_CWE is: {hashCodeAllowsNullCwe}")
         return hashCodeAllowsNullCwe
 
-    def delete(self, *args, **kwargs):
+    def delete(self, *args, product_grading_option=True, **kwargs):
         logger.debug("%d test delete", self.id)
         super().delete(*args, **kwargs)
-        with suppress(Test.DoesNotExist, Engagement.DoesNotExist, Product.DoesNotExist):
-            # Suppressing a potential issue created from async delete removing
-            # related objects in a separate task
-            calculate_grade(self.engagement.product)
+        if product_grading_option:
+            with suppress(Test.DoesNotExist, Engagement.DoesNotExist, Product.DoesNotExist):
+                # Suppressing a potential issue created from async delete removing
+                # related objects in a separate task
+                system_settings = System_Settings.objects.get()
```

Comment:
_slight_ optimization
```suggestion
                system_settings = System_Settings.objects.only("enable_product_grade").get()
```

Reply:
Product grade calculation is also performed on the engagement delete, but the check for system settings was not applied there